src/components/page-header/index.tsx

```diff
@@ -13,9 +13,22 @@ interface IPageHeader {
 
 const PageHeader = ({ children, border = true }: IPageHeader) => {
   const navigate = useNavigate()
+  const location = useLocation()
+
+  const isReservationListPage = location.pathname === '/reservation-list'
+  const isMyPage = location.pathname === '/mypage'
+  const handleClickBack = () => {
+    if (isReservationListPage) {
+      navigate('/mypage')
+    } else if (isMyPage) {
+      navigate('/')
+    } else {
+      navigate(-1)
+    }
+  }
```

Comment:
P4: 뒤로 가기 로직이 상황 따라 더 추가될 가능성을 고려했을 때,  페이지를 1번 뒤로 가는 로직을 default로 하고, 다른 로직은 객체로 분리해보면 어떨까요?
```suggestion
const navigationRoutes: Record<string, string> = {
    '/reservation-list': '/mypage',
    '/mypage': '/',
  };

  const handleClickBack = () => {
    const defaultRoute = -1;
    const route = navigationRoutes[location.pathname] || defaultRoute;
    navigate(route);
  };
```